### PR TITLE
[K8SPXC-783]: grant SYSTEM_USER for monitor and clustercheck users

### PR DESCRIPTION
**Pull request by defbin**

### CODE REVIEW

File: build/pxc-entrypoint.sh [hunk 1]
```diff
@@ -364,12 +364,12 @@ if [ -z "$CLUSTER_JOIN" ] && [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 			GRANT ALL ON *.* TO 'xtrabackup'@'%';
 
 			CREATE USER 'monitor'@'${MONITOR_HOST}' IDENTIFIED BY '${MONITOR_PASSWORD}' WITH MAX_USER_CONNECTIONS 100;
-			GRANT SELECT, PROCESS, SUPER, REPLICATION CLIENT, RELOAD ON *.* TO 'monitor'@'${MONITOR_HOST}';
+			GRANT SELECT, PROCESS, SUPER, REPLICATION CLIENT, RELOAD, SYSTEM_USER ON *.* TO 'monitor'@'${MONITOR_HOST}';
```
Comment (hors) [inline build/pxc-entrypoint.sh:367]: ok, but what about existing users?

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| storage | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| recreate | passed |
| scheduled-backup | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| init-deploy | failed |
| demand-backup | passed |
| smart-update | passed |
| limits | passed |
| self-healing-chaos | failed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| self-healing-advanced-chaos | failed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| operator-self-healing-chaos | failed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/f11582fb783b6cdc7976e42c1d11fb66aa261f80
image: `perconalab/percona-xtradb-cluster-operator:PR-994-f11582fb`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| security-context | passed |
| self-healing-chaos | failed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | passed |
| big-data | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| init-deploy | failed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |
| smart-update | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | failed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/f11582fb783b6cdc7976e42c1d11fb66aa261f80
image: `perconalab/percona-xtradb-cluster-operator:PR-994-f11582fb`

File: build/pxc-entrypoint.sh [hunk 2]
```diff
@@ -364,12 +364,12 @@ if [ -z "$CLUSTER_JOIN" ] && [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
...
 			GRANT SELECT ON performance_schema.* TO 'monitor'@'${MONITOR_HOST}';
 			${monitorConnectGrant}
 
 			CREATE USER 'clustercheck'@'localhost' IDENTIFIED BY '${CLUSTERCHECK_PASSWORD}';
-			GRANT PROCESS ON *.* TO 'clustercheck'@'localhost';
```
Comment (hors) [inline build/pxc-entrypoint.sh:372]: please re-check this user, maybe we don't use it at all now

Comment (defbin) [reply build/pxc-entrypoint.sh:372]: I'm not sure the user is not used anymore. I definitely see it in many tests

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | failed |
| haproxy | failed |
| scaling | failed |
| big-data | passed |
| scheduled-backup | failed |
| recreate | passed |
| storage | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | failed |
| upgrade-proxysql | failed |
| self-healing-chaos | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| limits | failed |
| monitoring-2-0 | passed |
| smart-update | passed |
| affinity | passed |
| self-healing-advanced-chaos | passed |
| one-pod | failed |
| scaling-proxysql | failed |
| auto-tuning | failed |
| pitr | failed |
| operator-self-healing | failed |
| upgrade-consistency | failed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | failed |
| security-context | failed |
| operator-self-healing-chaos | failed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/f138de873168f4584633af1844b40b8250cbdf2e
image: `perconalab/percona-xtradb-cluster-operator:PR-994-f138de87`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | passed |
| upgrade-haproxy | passed |
| storage | passed |
| recreate | passed |
| haproxy | passed |
| big-data | passed |
| self-healing | passed |
| init-deploy | failed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| demand-backup | passed |
| self-healing-chaos | failed |
| monitoring-2-0 | passed |
| smart-update | passed |
| self-healing-advanced | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| self-healing-advanced-chaos | failed |
| operator-self-healing | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing-chaos | failed |
| users | failed |
| upgrade-consistency | passed |
| demand-backup-encrypted-with-tls | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/26d9870d2a97bdb1a3225968fe6a06066d618ea2
image: `perconalab/percona-xtradb-cluster-operator:PR-994-26d9870d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | failed |
| upgrade-haproxy | passed |
| scaling | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| big-data | passed |
| storage | passed |
| scheduled-backup | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| upgrade-proxysql | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |
| smart-update | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | failed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| security-context | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/26d9870d2a97bdb1a3225968fe6a06066d618ea2
image: `perconalab/percona-xtradb-cluster-operator:PR-994-26d9870d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | passed |
| scaling | passed |
| big-data | passed |
| scaling-proxysql | passed |
| security-context | passed |
| storage | passed |
| upgrade-haproxy | passed |
| recreate | passed |
| self-healing | passed |
| haproxy | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-chaos | passed |
| init-deploy | failed |
| demand-backup | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/26d9870d2a97bdb1a3225968fe6a06066d618ea2
image: `perconalab/percona-xtradb-cluster-operator:PR-994-26d9870d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | failed |
| recreate | failed |
| upgrade-haproxy | failed |
| scheduled-backup | failed |
| big-data | failed |
| storage | failed |
| scaling | failed |
| init-deploy | failed |
| restore-to-encrypted-cluster | failed |
| scaling-proxysql | failed |
| self-healing | failed |
| upgrade-proxysql | failed |
| limits | failed |
| demand-backup | failed |
| security-context | failed |
| self-healing-chaos | failed |
| smart-update | failed |
| monitoring-2-0 | failed |
| pitr | failed |
| self-healing-advanced | failed |
| upgrade-consistency | failed |
| affinity | failed |
| demand-backup-encrypted-with-tls | failed |
| self-healing-advanced-chaos | failed |
| one-pod | failed |
| operator-self-healing | failed |
| auto-tuning | failed |
| operator-self-healing-chaos | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |
| validation-hook | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/e3ef1f3ae84d79a1977c801616c7a74db55400ab
image: `perconalab/percona-xtradb-cluster-operator:PR-994-e3ef1f3a`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/e3ef1f3ae84d79a1977c801616c7a74db55400ab
image: `perconalab/percona-xtradb-cluster-operator:PR-994-e3ef1f3a`

File: build/pxc-entrypoint.sh (hunk 2 above)
Comment (defbin) [reply build/pxc-entrypoint.sh:372]: K8SPXC-878

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scheduled-backup | passed |
| storage | passed |
| haproxy | passed |
| scaling | passed |
| recreate | passed |
| upgrade-haproxy | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| upgrade-proxysql | passed |
| limits | passed |
| self-healing-chaos | failed |
| security-context | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| self-healing-advanced | passed |
| smart-update | passed |
| affinity | passed |
| one-pod | passed |
| self-healing-advanced-chaos | failed |
| auto-tuning | passed |
| operator-self-healing | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| operator-self-healing-chaos | failed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| upgrade-consistency | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/e3ef1f3ae84d79a1977c801616c7a74db55400ab
image: `perconalab/percona-xtradb-cluster-operator:PR-994-e3ef1f3a`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| storage | passed |
| recreate | passed |
| scheduled-backup | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| pitr | passed |
| big-data | passed |
| scaling | passed |
| demand-backup-encrypted-with-tls | passed |
| haproxy | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/994/commits/e3ef1f3ae84d79a1977c801616c7a74db55400ab
image: `perconalab/percona-xtradb-cluster-operator:PR-994-e3ef1f3a`

**Review by nonemax — APPROVED** [review]
lgtm

**Review by delgod — APPROVED** [review]

**Pull request merged**